pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -113,6 +122,24 @@ func validateIPAddresses(a *cmapi.CertificateSpec, fldPath *field.Path) field.Er
 	return el
 }
 
+func validateEmailAddresses(a *cmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	if len(a.EmailSANs) <= 0 {
+		return nil
+	}
+	el := field.ErrorList{}
+	for i, d := range a.EmailSANs {
+		e, err := mail.ParseAddress(d)
+		if err != nil {
+			el = append(el, field.Invalid(fldPath.Child("emailSANs").Index(i), d, "invalid email address"))
+		} else if e.Address != d {
+			// Go accepts email names as per RFC 5322 (name <email>)
+			// This checks if the supplied value only contains the email address and nothing else
+			el = append(el, field.Invalid(fldPath.Child("emailSANs").Index(i), d, "invalid email address"))
```

Comment:
Can we provide some additional info to the user here aside from just "invalid"? i.e. an example of what _is_ valid, or an explanation of what is invalid?

In the `if err != nil` case, we definitely have additional information we can display (`err`), and in this branch we should be able to provide an example of what is valid (given the user must have entered something a bit weird to get into this state)